controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -24,6 +24,8 @@ def setUp(self):
         self.detect_array_type = detect_array_type_patcher.start()
         self.detect_array_type.return_value = "a9k"
         self.addCleanup(detect_array_type_patcher.stop)
+        self.request = Mock()

```

Comment:
I am not sure it is accurate to put it on the global setup since several tests change the request (an not all the information in the request).
For example: self.request.name, self.request.source_volume_id, self.request.parameters, self.request.volume_content_source, etc.
When Mock called each setup it doesn't contain leftovers which might affect the test.
(Not all setUp methods call to super().setUp)

Reply:
a. all `setUp` methods that inherit from `AbstractControllerTest` calls to `super().setUp`.
b. we are not using a `self.request` parameter that we are not config to our needs or at least knows what its value is..